### PR TITLE
Spreading and compositing operators

### PR DESCRIPTION
**Pull request by jcrist**
- Adds compositing operators from Porter/Duff paper, used for merging multiple pixels into a single value
- Adds spreading function for spreading pixels in a shape. This is useful for making sparse images more visible.

Todo:

- [x] Tests
- [x] Fix merge/stack to use compositing operators
- [x] Docs

### CODE REVIEW

File: datashader/transfer_functions.py [hunk 1]
```diff
@@ -163,3 +166,144 @@ def colorize(agg, color_key, how='cbrt', min_alpha=20):
...
+    while y < r:
+        _fill_span(arr, y0 + y, x0 - x, x0 + x, val)
+        _fill_span(arr, y0 - y, x0 - x, x0 + x, val)
+        y += 1
+        if do2 <= 0:
+            do2 += 2 * y + 1
+        else:
+            x -= 1
+            do2 += 2 * (y - x) + 1
+
```
Comment (jbednar) [inline datashader/transfer_functions.py]: Just curious -- this looks like a pretty complicated implementation; what does it buy over something more like:

```python
>>> import numpy as np                                                                                                                                  
>>> def disk(x, y, radius):                                                                                                                             
      distance_from_origin = np.sqrt(x**2+y**2)                                                                                                         
      distance_outside_disk = distance_from_origin - radius                                                                                             
      return np.where(distance_from_origin - radius<=0,1.0,0.0)
>>> x,y = np.mgrid[-5:5, -5:5] * 0.1
>>> disk(x,y,0.2)
array([[ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  1.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  1.,  1.,  1.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  1.,  1.,  1.,  1.,  1.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  1.,  1.,  1.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  1.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.]])
```

**Comment by jbednar** [comment]
Looks great, and really useful!

**Comment by jcrist** [comment]
This is ready for review.

**Comment by jcrist** [comment]
Here's a notebook showing this in use: https://gist.github.com/jcrist/62b366727886561356d8

**Comment by jbednar** [comment]
Looks great! The new code is clean and clear, and seems very useful.  Merging.

**Pull request merged**